docs/tutorials/data_provider.md

```diff
@@ -64,25 +64,44 @@ impl AdditiveIdentity {
 
 ## Caching Data Provider
 
-It is not generally required to implement a cache, especially if using the BakedDataProvider. However, if you wish to implement a cache in your data pipeline, you can do so like this:
+ICU4X has no internal caches because there is no one-size-fits-all solution. It is easy for clients to implement their own cache for ICU4X, and although this is not generally required or recommended, it may be beneficial when latency is of utmost importance and:
+
+1. A less-efficient data provider is being used, such as JSON
+2. Lookups make heavy use of vertical fallback
```

Comment:
thought: I don't think we should suggest caching the data structs when fallback is the problem. In that case one should cache resolved locales

Reply:
The resolved locales could be different for each key. But yes, we could cache `(DataKey, DataLocale) -> DataLocale`.

Reply:
I removed the use case of vertical fallback since this is a good point and we could even consider offering this as an official solution if needed.